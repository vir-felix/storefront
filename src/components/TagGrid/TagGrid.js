import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "material-ui/styles";
import Chip from "material-ui/Chip";
import Typography from "material-ui/Typography";
import Grid from "material-ui/Grid";
import Link from "components/Link";

const styles = (theme) => ({
  title: {
    marginBottom: theme.spacing.unit
  },
  chip: {
    cursor: "pointer"
  }
});

/**
 * Tag grid - displays a grid of tags
 */
@withStyles(styles, { withTheme: true })
export default class TagGrid extends Component {
  static propTypes = {
    /**
     * CSS class names
     */
    classes: PropTypes.object,

    /**
     * Array of tag nodes [{ node: Object }]
     */
    tags: PropTypes.array,

    /**
     * Theme
     */
    theme: PropTypes.object
  }

  render() {
    const {
      classes,
      tags,
      theme
    } = this.props;


    if (!Array.isArray(tags)) return null;

    return (
      <section>
        <Typography className={classes.title} variant="title">{"Tags"}</Typography>
        <Grid container spacing={theme.spacing.unit}>
          {tags.map(({ node: tag }, index) => (
            <Grid item>
              <Link route={`/tag/${tag.slug}`}>
                <Chip className={classes.chip} key={index} label={tag.name} />
              </Link>
            </Grid>
          ))}
        </Grid>
      </section>
    );
  }
}
