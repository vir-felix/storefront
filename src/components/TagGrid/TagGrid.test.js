import React from "react";
import renderer from "react-test-renderer";
import TagGrid from "./TagGrid";

const tags = [
  {
    node: {
      position: null,
      name: "Shoes",
      slug: "shoes"
    }
  },
  {
    node: {
      position: null,
      name: "Shop",
      slug: "shop"
    }
  }
];

test("basic snapshot", () => {
  const component = renderer.create((
    <TagGrid tags={tags} />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
