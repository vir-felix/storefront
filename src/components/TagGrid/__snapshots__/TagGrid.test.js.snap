// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<section>
  <h2
    className="MuiTypography-root-3 MuiTypography-title-9 TagGrid-title-1"
  >
    Tags
  </h2>
  <div
    className="MuiGrid-container-29 MuiGrid-spacing-xs-8-51"
  >
    <div
      className="MuiGrid-item-30"
    >
      <a
        className="Link-anchor-126"
        href="/tag/shoes"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <div
          className="MuiChip-root-127 TagGrid-chip-2"
          onClick={undefined}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiChip-label-146"
          >
            Shoes
          </span>
        </div>
      </a>
    </div>
    <div
      className="MuiGrid-item-30"
    >
      <a
        className="Link-anchor-126"
        href="/tag/shop"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <div
          className="MuiChip-root-127 TagGrid-chip-2"
          onClick={undefined}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiChip-label-146"
          >
            Shop
          </span>
        </div>
      </a>
    </div>
  </div>
</section>
`;
